kubetest/kind/kind.go

```diff
@@ -335,28 +335,8 @@ func (d *Deployer) Up() error {
 func (d *Deployer) IsUp() error {
 	log.Println("kind.go:IsUp()")
 
-	// Proceed only if a cluster exists.
-	exists, err := d.clusterExists()
-	if err != nil {
-		return err
-	}
-	if !exists {
-		log.Printf("kind.go:IsUp(): no such cluster %q; skipping IsUp()!", d.kindClusterName)
-		return nil
-	}
-
-	// Obtain the path of the kubeconfig.
-	path, err := d.getKubeConfigPath()
-	if err != nil {
-		return err
-	}
-
-	// Check if kubectl reports nodes for that kubeconfig file.
+	// Check if kubectl reports nodes.
 	cmd, err := d.KubectlCommand()
```

Comment:
do we ned to set the kubeconfig?

Reply:
Looks like yes

Reply:
so we set it at Up(), but to me it seems like this first IsUp() call if it returns nil it skips Up(), according to the log.

Reply:
check the log here: https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kind-master/220

Reply:
uh lol the job doesn't have `--up`
```yaml
apiVersion: prow.k8s.io/v1
kind: ProwJob
metadata:
  annotations:
    prow.k8s.io/job: ci-kubernetes-e2e-kubeadm-kind-master
  creationTimestamp: null
  labels:
    created-by-prow: "true"
    preset-bazel-remote-cache-enabled: "true"
    preset-bazel-scratch-dir: "true"
    preset-dind-enabled: "true"
    prow.k8s.io/id: 9dd7c809-322e-11e9-aa24-0a580a6c019d
    prow.k8s.io/job: ci-kubernetes-e2e-kubeadm-kind-master
    prow.k8s.io/type: periodic
  name: 90d351f5-3234-11e9-89b1-0a580a6c061b
spec:
  agent: kubernetes
  cluster: default
  decoration_config:
    gcs_configuration:
      bucket: kubernetes-jenkins
      default_org: kubernetes
      default_repo: kubernetes
      path_strategy: legacy
    gcs_credentials_secret: service-account
    grace_period: 15000000000
    timeout: 2400000000000
    utility_images:
      clonerefs: gcr.io/k8s-prow/clonerefs:v20190125-2aca69d
      entrypoint: gcr.io/k8s-prow/entrypoint:v20190125-2aca69d
      initupload: gcr.io/k8s-prow/initupload:v20190125-2aca69d
      sidecar: gcr.io/k8s-prow/sidecar:v20190125-2aca69d
  extra_refs:
  - base_ref: master
    org: kubernetes
    path_alias: k8s.io/kubernetes
    repo: kubernetes
  - base_ref: master
    org: kubernetes
    path_alias: k8s.io/kubeadm
    repo: kubeadm
  - base_ref: master
    org: kubernetes-sigs
    path_alias: sigs.k8s.io/kind
    repo: kind
  job: ci-kubernetes-e2e-kubeadm-kind-master
  namespace: test-pods
  pod_spec:
    containers:
    - args:
      - --build=bazel
      - --provider=skeleton
      - --deployment=kind
      - --kind-binary-version=build
      - --kind-config-path=./../../k8s.io/kubeadm/tests/e2e/kind/single-cp/single-cp.yaml
      - --test
      - --check-version-skew=false
      - --test_args=--ginkgo.focus=\[Conformance\] --ginkgo.skip=\[Serial\]|Alpha|Kubectl|\[(Disruptive|Feature:[^\]]+|Flaky)\]
        --disable-log-dump=true --num-nodes=3
      - --ginkgo-parallel
      - --timeout=30m
      command:
      - runner.sh
      - kubetest
      env:
      - name: REPO_OWNER
        value: kubernetes
      - name: REPO_NAME
        value: kubernetes
      - name: TEST_TMPDIR
        value: /bazel-scratch/.cache/bazel
      - name: BAZEL_REMOTE_CACHE_ENABLED
        value: "true"
      - name: DOCKER_IN_DOCKER_ENABLED
        value: "true"
      image: gcr.io/k8s-testimages/kubekins-e2e:latest-master
      imagePullPolicy: Always
      name: ""
      resources:
        requests:
          cpu: "2"
          memory: 9000Mi
      securityContext:
        privileged: true
      volumeMounts:
      - mountPath: /lib/modules
        name: modules
        readOnly: true
      - mountPath: /sys/fs/cgroup
        name: cgroup
      - mountPath: /bazel-scratch/.cache
        name: bazel-scratch
      - mountPath: /docker-graph
        name: docker-graph
    volumes:
    - hostPath:
        path: /lib/modules
        type: Directory
      name: modules
    - hostPath:
        path: /sys/fs/cgroup
        type: Directory
      name: cgroup
    - emptyDir: {}
      name: bazel-scratch
    - emptyDir: {}
      name: docker-graph
  type: periodic
status:
  startTime: "2019-02-16T21:48:16Z"
  state: triggered
```

Reply:
it failed with or without this change in the same way.
i'm trying to figure out why this IsUp() is called in the first place.

the two location where a deployer's IsUp() can be called:
https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L576
kubemark seems unrelated here?

https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L177
if --test is passed, but at this point Up() was not called according to the log.

in terms of the job aren't we supposed to add --up and --down as args too?
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml#L38-L48

Reply:
https://github.com/kubernetes/test-infra/pull/11335#issuecomment-464389844

Reply:
ok, thanks for the PR.